### ISSUE TITLE
Commit verification should not fail on single signature failure

### ISSUE THREAD

**Issue by Frank-Buss** [cometbft/cometbft#1823]
In current master, verifyCommitSingle returns with an error, as soon as the signature of one validator can't be verified, see here:
https://github.com/cometbft/cometbft/blob/437391a0cb766ec0e7a2d87242a90b271ac31707/types/validation.go#L373
This defeats the intended concept that only 2/3 of all validators must be right. It allows easy denial of service attacks, if someone controls only one node. Incidentally needs to be one of the first 2/3 validators, because of the early return when 2/3 is reached, but still wrong according to the specification, and the attack is possible if someone controls for example only the first validator.

Should be easy to fix: talliedVotingPower needs to be skipped if a validator is wrong, instead of returning with an error. Probably good idea to add nice logging which validator was wrong.

**Comment by melekes** [comment]
> It allows easy denial of service attacks, if someone controls only one node.

Were you able to simulate the attack?

If the attacker controls one validator, it can send a vote with an invalid signature, which an honest proposer won't include. Now, if the proposer is the rogue node itself, then it can create a last commit with the invalid signature, but then again the proposed block won't be accepted by the honest majority ([1](https://github.com/cometbft/cometbft/blob/437391a0cb766ec0e7a2d87242a90b271ac31707/internal/state/validation.go#L92)).

**Comment by sergio-mena** [comment]
I agree with @melekes's reasoning, and would like to expand on it.

@Frank-Buss The line you point to is part of the verification of a commit. In order to construct such a commit, any non-byzantine node should have checked the signature of all votes in advance as they were received from the network. If a signature was invalid, it should have been discarded at that point. The Tendermint algorithm would force the honest node to wait for further votes until 2/3 were received with correct signatures.

In conclusion, if a commit contains an invalid signature, as @melekes says, it _must_ have been constructed by a byzantine node and must therefore be discarded altogether.

**Comment by andynog** [comment]
Hi @Frank-Buss, please let us know if the explanations provided by @melekes and @sergio-mena are sufficient. As this seems not to be a bug but the right construct by design please let us know if it's OK to go ahead and close this issue. Thanks !

**Comment by Frank-Buss** [comment]
Thanks for your comments. I'm waiting for a response from a colleague, because he thought it could be a problem in our system where we use it, and he has more experience with the blockchain, and will then close it, if he agrees.

**Comment by Frank-Buss** [comment]
We discussed it in our team, and we agree with your reasoning. We had this problem with the invalid signature sometimes, but this is probably a different problem in our implementation with regard to empty signatures.
